### PR TITLE
OpenMp Egg

### PR DESCRIPTION
**Pull request by vedran77**
# Description

This pull request adds the OpenMp egg to the Pterodactyl panel, enabling easy deployment and management of game servers for a modified version of GTA San Andreas. This modification, inspired by SAMP (San Andreas Multiplayer), introduces unique features and gameplay enhancements.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [ ] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [ ] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [ ] The egg was exported from the panel

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [x] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/gta/openmp/egg-open-m-p.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        "startup": "{}",
+        "logs": "{}",
+        "stop": "^^C"
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\r\n\r\necho \"running: curl -sSL -o openmp.tar.gz https:\/\/github.com\/openmultiplayer\/open.mp\/releases\/download\/v1-RC2\/open.mp-linux-x86-dynssl.tar.gz\"\r\ncurl -sSL -o openmp.tar.gz https:\/\/github.com\/openmultiplayer\/open.mp\/releases\/download\/v1-RC2\/open.mp-linux-x86-dynssl.tar.gz\r\n\r\nmkdir -p \/mnt\/server\r\ntar -xzvf openmp.tar.gz --strip-components=1 -C \/mnt\/server\/\r\n\r\ncd \/mnt\/server || exit\r\n\r\necho \"creating default config\"\r\n\r\nif [ -e conf.json ]; then\r\n    echo \"server config file exists\"\r\nelse\r\n    echo \"Downloading default OpenMp config\"\r\n    curl https:\/\/raw.githubusercontent.com\/parkervcp\/eggs\/master\/game_eggs\/gta\/openmp\/confing.json >> confing.json\r\nfi\r\n\r\nchown -R root:root \/mnt\r\n\r\nexport HOME=\/mnt\/server\r\n\r\n\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (QuintenQVD0) [inline game_eggs/gta/openmp/egg-open-m-p.json:25]: Please do not hardcode the sersion but use our github release grab script: https://github.com/parkervcp/eggs/blob/master/scripts/github_release_grabber.sh

This:
```bash
chown -R root:root /mnt

export HOME=/mnt/server
```
Is not needed as that will be auto done by the wings

File: game_eggs/gta/openmp/egg-open-m-p.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            "container": "ghcr.io\/parkervcp\/installers:debian",
+            "entrypoint": "bash"
+        }
+    },
+    "variables": [
+        {
+            "name": "RCON_PASSWORD",
+            "description": "RCON_PASSWORD",
+            "env_variable": "RCON_PASSWORD",
+            "default_value": "rcon_pass",
```
Comment (QuintenQVD0) [inline game_eggs/gta/openmp/egg-open-m-p.json:35]: if thi is required then make the default value empty for sequrity reasons and should you realy limit this to 20 wy not 32 or 64?

File: game_eggs/gta/openmp/config.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
+{
```
Comment (QuintenQVD0) [inline game_eggs/gta/openmp/config.json:1]: is there any reason this is on .json wile there docs mention .cfg?
https://www.open.mp/docs/server/server.cfg

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/gta/openmp/config.json (hunk 3 above)
Comment (vedran77) [reply game_eggs/gta/openmp/config.json:1]: Docs are for SA-MP, maybe somewhere on open.mp there is documentation for config.json, but open.mp server won't work without config.json, server.cfg isn't supported

File: game_eggs/gta/openmp/egg-open-m-p.json (hunk 2 above)
Comment (vedran77) [reply game_eggs/gta/openmp/egg-open-m-p.json:35]: I will put "changeme", server won't start if that password is "changeme".

Comment (QuintenQVD0) [reply game_eggs/gta/openmp/egg-open-m-p.json]: Then leave it emty and force the user to set it

**Comment by QuintenQVD0** [comment]
...you can not make changes to the .json file directly. You have to fully export it. It is even on the top as a comment in the json file

**Review by QuintenQVD0 — APPROVED** [review]

**Pull request merged**